main.rb

```diff
@@ -0,0 +1,158 @@
+# rubocop: disable Metrics/ModuleLength
+# rubocop: disable Metrics/PerceivedComplexity, Metrics/CyclomaticComplexity
+module Enumerable
+  def my_each
+    return to_enum(:my_each) unless block_given?
+
+    i = 0
+    arr = to_a if self.class == Hash || self.class == Range
+    while i < size
+      if self.class == Hash || self.class == Range
+        yield(arr.at(i))
+      else
+        yield(at(i))
+      end
+      i += 1
+    end
+    self
+  end
+
+  def my_each_with_index(*arg)
+    return to_enum(:my_each_with_index) unless block_given? && arg.empty?
+
+    index = 0
+    arr = self
+    tarr = arr.to_a
+    i = 0
+    while i < arr.size
+      yield(tarr[i], index)
+      index += 1
+      i += 1
+    end
+    self
+  end
+
+  def my_select
+    return to_enum(:my_select) unless block_given?
+
+    item = []
+    arr = self
+    tarr = []
+    if arr.class == Hash
+      arr.my_each { |key, _value| tarr << key }
+    else
+      tarr = arr.to_a
+    end
+    i = 0
+    while i < tarr.size
+      yield(tarr[i]) ? item << tarr[i] : nil
+      i += 1
+    end
+    item
+  end
+
+  def my_all?(*args)
+    flag = 0
+    i = 0
+    arr = self
+    is_a?(Range) || is_a?(Hash) ? arr = arr.to_a : nil
+
+    case args.empty?
+    when true
+      if block_given?
+        blk = yield(arr.at(i))
+        arr.my_each { flag += 1 if blk == true }
+      else
+        arr.my_each { flag += 1 unless i.nil? || i == false }
+      end
+    when false
+      arr.my_each { flag += 1 if i.is_a?(args[0]) }
+    end
+    return true if flag == size
+    return false if flag != size
+  end
+
+  def my_any?(*args)
+    flag = 0
+    case args.empty?
+    when true
+      if block_given?
+        my_each { |i| flag += 1 if yield(i) }
+      else
+        my_each { |i| flag += 1 unless i.nil? || i == false }
+      end
+    when false
+      my_each { |i| flag += 1 if i.is_a?(args[0]) }
+    end
+    return true if flag.positive?
+    return false if flag <= 0
+  end
+
+  def my_none?(*args)
+    counter = 0
+    case !block_given?
+    when true
+      if args.empty?
+        my_each { |i| counter += 1 if i != true }
+      else
+        my_each { |i| counter += 1 if i == args[0] || i.is_a?(args[0]) }
+      end
+    when false
+      my_each { |i| counter += 1 if yield(i) == true }
+    end
+    return true if counter.zero?
+    return false if counter.positive?
+  end
```

Comment:
# my_none? 
1. [ ] When no block or argument is given returns true only if none of the collection members is true

      Failure/Error: expect(false_array.my_none?).to be true

        expected true
             got false
##
# my_none? 
2. [ ] When a Regex is passed as an argument returns true if none of the collection matches the Regex

      Failure/Error: expect(words.my_none?(/z/)).to be words.none?(/z/)

      TypeError:
        class or module required
##
# my_none? 
3. [ ] When a Regex is passed as an argument returns false if any of the collection matches the Regex

      Failure/Error: expect(words.my_none?(/d/)).to be words.none?(/d/)

      TypeError:
        class or module required
##
# my_none? 
4. [ ] When a pattern other than Regex or a Class is given returns true only if none of the collection matches the pattern

      Failure/Error: expect(words.my_none?(5)).to be words.none?(5)

      TypeError:
        class or module required
##    
# my_none? 
5. [ ] When a pattern other than Regex or a Class is given returns false only if one of the collection matches the pattern

      Failure/Error: expect(words.my_none?(5)).to be words.none?(5)

      TypeError:
        class or module required